### PR TITLE
Python 3: Port some html tests following Python file handlers p…

### PR DESCRIPTION
**Pull request by ziransun**
…orting guide

This is a rework on https://github.com/web-platform-tests/wpt/pull/22402
This change follows the porting guide discussed in RFC49:
https://github.com/web-platform-tests/rfcs/pull/49

### CODE REVIEW

**Comment by ziransun** [comment]
@Hexcles @stephenmcgruer: when you find time, is it okay to have  a look? There are a few places I'm not so sure. They were commented with "TBC:". Many thanks!

File: xhr/resources/inspect-headers.py
Comment (ziransun) [inline xhr/resources/inspect-headers.py:12]: I might need to rethink about this. Please ignore this line for now.

File: cookies/resources/set-cookie.py
Comment (Hexcles) [inline cookies/resources/set-cookie.py:26]: I'm not sure if this is supposed to be `utf-8`.

After #23284 , this whole handler could be refactored to use `request.GET["name"]` and `request.GET["path"]`, both of which are already binary, instead of parsing the query again.

**Review by Hexcles — COMMENTED** [review]
Less than half way through but it's already clear that we really need to get #23284 merged...

File: xhr/resources/inspect-headers.py
Comment (Hexcles) [inline xhr/resources/inspect-headers.py:3]: It's unfortunate that this function uses `raw_headers`. It'd be great if we could refactor this to avoid using `raw_headers`. It's not obvious to me why it'd need to do that. Maybe tag someone familiar with xhr to confirm.

**Review by Hexcles — APPROVED** [review]
LGTM. This is basically what I expected. Please wait for review by @annevk , @jgraham , etc.

File: cookies/resources/set-cookie.py
Comment (jgraham) [inline cookies/resources/set-cookie.py:22]: Yoo can still use a format string with bytes in both Python 2 and 3

File: html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py
Comment (jgraham) [inline html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: The response bodies don't have to be bytes.

File: xhr/resources/inspect-headers.py
Comment (jgraham) [reply xhr/resources/inspect-headers.py:3]: @annevk might want to comment here

Comment (annevk) [reply xhr/resources/inspect-headers.py:3]: How would you do the kind of checks it does without raw headers?

Comment (annevk) [reply xhr/resources/inspect-headers.py:3]: raw_headers is important and I hope we're not refactoring it away in places where it has been used, as we'll lose coverage.

File: html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py
Comment (annevk) [inline html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py:4]: Does this mean that not all strings need to be annotated?

File: xhr/resources/inspect-headers.py
Comment (Hexcles) [reply xhr/resources/inspect-headers.py:3]: It won't go away, but as you can see, this kind of encoding juggling will be needed wherever it's used.

IIUC, this function only needs to filter by header values or names. The former could be implemented by going through `request.headers.itervalues()` while the latter with `request.headers.get_list(filter_name)`.

File: html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py
Comment (Hexcles) [reply html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py:4]: Good catch. We plan to annotate all strings and I missed this one.

File: xhr/resources/inspect-headers.py
Comment (annevk) [reply xhr/resources/inspect-headers.py:3]: No, this function cares about the exact way headers are serialized. E.g., that there's a space after the colon. That they end with CRLF. It also cares about the header name not changing case. I very much doubt the higher level libraries enforce that as it's not something H/1 really enforces either way.

Comment (Hexcles) [reply xhr/resources/inspect-headers.py:3]: I see. If the test needs to explicitly assert the syntax, then yes we'd have to do this. The parsing logic in stdlib is much laxer.

Comment (annevk) [reply xhr/resources/inspect-headers.py:3]: That's what it does by doing its own parsing, right? I think this caught an issue browsers wanted to align on some years ago, but it could have been another usage of raw_headers as well.

File: cookies/resources/set-cookie.py
Comment (ziransun) [reply cookies/resources/set-cookie.py:22]: Is it okay to explain a bit more what you mean here? Thank you.

The code here was refactored as I wasn't sure what encoding scheme to use to convert string to bytes after the .format() called in the original code.

File: html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py
Comment (ziransun) [reply html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: Yes. My thought was to make the type of returned response body consistent with the rest...

File: cookies/resources/set-cookie.py
Comment (stephenmcgruer) [reply cookies/resources/set-cookie.py:22]: `b"%b=1; Path=%b; Expires=Wed, 09 Jun 2021 10:18:14 GMT" % (name, path)` works, I believe

File: html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py
Comment (stephenmcgruer) [reply html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: cc @Hexcles - what do you think we should do about this sort of situation?

Comment (Hexcles) [reply html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: Aesthetically, I'm with @ziransun here, i.e. slightly leaning towards `b""` in this return form.

If it's e.g. `return ""`, I'd say either is completely fine; and I'm not sure if we need to define a guideline on which type to prefer for this case (even if we do, it'd be kind of hard to enforce).

Comment (annevk) [reply html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: I think at a minimum we have to support byte bodies. Running UTF-8 encode on `u""` strings when given seems fine though as a convenience.

Comment (Hexcles) [reply html/semantics/scripting-1/the-script-element/module/resources/check-cookie.py:13]: Yes we support both.

File: cookies/resources/set-cookie.py
Comment (Hexcles) [reply cookies/resources/set-cookie.py:22]: `b"%s=1; Path=%s; Expires=Wed, 09 Jun 2021 10:18:14 GMT" % (name, path)` does work in both. According to [PEP 461](https://www.python.org/dev/peps/pep-0461/) though, 

> %s is included as a synonym for %b for the sole purpose of making 2/3 code bases easier to maintain. Python 3 only code should use %b.

File: html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py
Comment (Hexcles) [inline html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py:14]: Alright, after some investigation, this should be **text**.

```suggestion
            response.set_error(404, u"Tried to read data not yet set")
```

Comment (Hexcles) [reply html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py:14]: Sent a PR to clarify the docstring #23555

File: html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py
Comment (Hexcles) [inline html/semantics/embedded-content/media-elements/track/track-element/cors/support/cors-tester.py:47]:
```suggestion
        return b"""WEBVTT
```

File: cookies/resources/set-cookie.py
Comment (Hexcles) [reply cookies/resources/set-cookie.py:22]:
```suggestion
    cookie = b"%s=1; Path=%s; Expires=Wed, 09 Jun 2021 10:18:14 GMT" % (name, path)
```

**Comment by Hexcles** [comment]
Alright, I can confirm `/preload/modulepreload.html` is already slightly flaky on master without this change.

**Comment by stephenmcgruer** [comment]
The RFC has now been merged. As far as I know this PR is approved by all the relevant parties, so admin-mergin (the TC failure is an existing flaky test).

**Pull request merged**
